### DISCUSSION TITLE
Azure DevOps connection problem

### DISCUSSION THREAD

**Discussion by sabine-ll** [six7/figma-tokens#1539] [Q&A]
Hi, has anyone experienced issue with connecting to  Azure DevOps. I have previously connected to GitLab without issues. However we are having no luck connecting to Azure. We tried many different combinations of URLs and access levels in Azure. The error we are getting is 'There is no branch' even though the main branch is there. Any help or advice is greatly appreciated. Thank you!

**Comment by sabine-ll** [comment]
Hi I'm still looking for some feedback. Has anyone successfully connected to Azure. I tried the credentials in another app and it worked fine. So wondering if there is a bug? Help is much appreciated. thanks

**Comment by JpG771** [comment]
Here is the 3 things I did to make it work.

1- Use server name, do not include repository name in url : https://my-server.com/Project/
Not : https://my-server.com/Project/_git/MyRepos
2- Use ending slash or a "undefined" string is added to the url like that : https://my-server.com/Project/
Not : https://my-server.com/Project or you will end with this https://my-server.com/ProjectUndefinded
3- Create a main branch in your repos. This is only for the request to be successful as figma seem to use it instead of master. Your commit will still go into your specified branch.

**Comment by mjdenton** [comment]
@sabine-ll, might seen obvious, but double check each of your inputs include no leading whitespaces. I've just had issues with Azure sync and it was due to a leading space having copied the branch name across from the Azure site.

**Comment by sefrijn** [comment]
Also ran into this issue with `There is no branch` alert.
The logging in the plugin should/could be improved. It had nothing to do with branches.

My solution:
Access tokens with limited permissions didn't work with me. So I added a DevOps token with full rights, and the plugin worked.
